``sys.argv`` is inferred wrongly as pylint's own values for ``sys.argv`` instead of ``Uninferrable``

### Bug description

Make file `t.py`
```
import sys

a, b, c = sys.argv

print(a, b, c)
```
and run `python t.py 1 2`

### Configuration

_No response_

### Command used

```shell
pylint t.py
```


### Pylint output

```shell
$ pylint t.py
************* Module t
t.py:1:0: C0114: Missing module docstring (missing-module-docstring)
t.py:3:0: W0632: Possible unbalanced tuple unpacking with sequence: left side has 3 label(s), right side has 2 value(s) (unbalanced-tuple-unpacking)

------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 0.00/10, +3.33)
```


### Expected behavior

The W0632 on line 3 looks like a bug, as in this case `sys.argv` is `['t.py', '1', '2']`

### Pylint version

```shell
$ pylint --version
pylint 2.15.5
astroid 2.12.12
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0]
```


### OS / Environment

```
$ uname -a
Linux DESKTOP-U8OKFP3 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7710.bugfix
+++ b/doc/whatsnew/fragments/7710.bugfix
@@ -1,0 +1,4 @@
+``sys.argv`` is now always correctly considered as impossible to infer (instead of
+using the actual values given to pylint).
+
+Closes #7710

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -41,7 +41,7 @@ dependencies    = [
     # Also upgrade requirements_test_min.txt.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/pylint-dev/astroid/issues/1341
-    "astroid>=3.0.0a9,<=3.1.0-dev0",
+    "astroid>=3.0.0b0,<=3.1.0-dev0",
     "isort>=4.2.5,<6",
     "mccabe>=0.6,<0.8",
     "tomli>=1.1.0;python_version<'3.11'",

--- a/requirements_test_min.txt
+++ b/requirements_test_min.txt
@@ -1,6 +1,6 @@
 .[testutils,spelling]
 # astroid dependency is also defined in pyproject.toml
-astroid==3.0.0a9  # Pinned to a specific version for tests
+astroid==3.0.0b0  # Pinned to a specific version for tests
 typing-extensions~=4.8
 py~=1.11.0
 pytest~=7.4

--- a/tests/functional/u/unbalanced/unbalanced_tuple_unpacking.py
+++ b/tests/functional/u/unbalanced/unbalanced_tuple_unpacking.py
@@ -1,5 +1,7 @@
 """Check possible unbalanced tuple unpacking """
 from __future__ import absolute_import
+
+import sys
 from typing import NamedTuple
 from functional.u.unpacking.unpacking import unpack
 
@@ -160,3 +162,8 @@ d, *_ = my_function("12")
 
 # https://github.com/pylint-dev/pylint/issues/5998
 x, y, z = (1, 2)  # [unbalanced-tuple-unpacking]
+
+# https://github.com/pylint-dev/pylint/issues/7710
+# Using a lot of args, so we have a high probability to still trigger the problem if
+# we add arguments to our unittest command later
+(p, q, r, s, t, u, v, w, x, y, z) = sys.argv

--- a/tests/functional/u/unbalanced/unbalanced_tuple_unpacking.txt
+++ b/tests/functional/u/unbalanced/unbalanced_tuple_unpacking.txt
@@ -1,9 +1,9 @@
-unbalanced-tuple-unpacking:11:4:11:27:do_stuff:"Possible unbalanced tuple unpacking with sequence '(1, 2, 3)': left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:17:4:17:29:do_stuff1:"Possible unbalanced tuple unpacking with sequence '[1, 2, 3]': left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:23:4:23:29:do_stuff2:"Possible unbalanced tuple unpacking with sequence '(1, 2, 3)': left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:82:4:82:28:do_stuff9:"Possible unbalanced tuple unpacking with sequence defined at line 7 of functional.u.unpacking.unpacking: left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:96:8:96:33:UnbalancedUnpacking.test:"Possible unbalanced tuple unpacking with sequence defined at line 7 of functional.u.unpacking.unpacking: left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:140:8:140:43:MyClass.sum_unpack_3_into_4:"Possible unbalanced tuple unpacking with sequence defined at line 128: left side has 4 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:145:8:145:28:MyClass.sum_unpack_3_into_2:"Possible unbalanced tuple unpacking with sequence defined at line 128: left side has 2 labels, right side has 3 values":INFERENCE
-unbalanced-tuple-unpacking:157:0:157:24::"Possible unbalanced tuple unpacking with sequence defined at line 151: left side has 2 labels, right side has 0 values":INFERENCE
-unbalanced-tuple-unpacking:162:0:162:16::"Possible unbalanced tuple unpacking with sequence '(1, 2)': left side has 3 labels, right side has 2 values":INFERENCE
+unbalanced-tuple-unpacking:13:4:13:27:do_stuff:"Possible unbalanced tuple unpacking with sequence '(1, 2, 3)': left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:19:4:19:29:do_stuff1:"Possible unbalanced tuple unpacking with sequence '[1, 2, 3]': left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:25:4:25:29:do_stuff2:"Possible unbalanced tuple unpacking with sequence '(1, 2, 3)': left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:84:4:84:28:do_stuff9:"Possible unbalanced tuple unpacking with sequence defined at line 7 of functional.u.unpacking.unpacking: left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:98:8:98:33:UnbalancedUnpacking.test:"Possible unbalanced tuple unpacking with sequence defined at line 7 of functional.u.unpacking.unpacking: left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:142:8:142:43:MyClass.sum_unpack_3_into_4:"Possible unbalanced tuple unpacking with sequence defined at line 130: left side has 4 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:147:8:147:28:MyClass.sum_unpack_3_into_2:"Possible unbalanced tuple unpacking with sequence defined at line 130: left side has 2 labels, right side has 3 values":INFERENCE
+unbalanced-tuple-unpacking:159:0:159:24::"Possible unbalanced tuple unpacking with sequence defined at line 153: left side has 2 labels, right side has 0 values":INFERENCE
+unbalanced-tuple-unpacking:164:0:164:16::"Possible unbalanced tuple unpacking with sequence '(1, 2)': left side has 3 labels, right side has 2 values":INFERENCE

